app/controlplane/plugins/core/webhook/v1/webhook.go

```diff
@@ -0,0 +1,321 @@
+}
+
+// registrationRequest defines the configuration required during registration
+type registrationRequest struct {
+	URL string `json:"url" jsonschema:"minLength=1,description=Webhook URL to send payloads to"`
+}
+
+// attachmentRequest defines the configuration required during attachment
+type attachmentRequest struct {
+	SendAttestation *bool `json:"send_attestation,omitempty" jsonschema:"description=Send attestation (default: true)"`
+	SendSBOM        *bool `json:"send_sbom,omitempty" jsonschema:"description=Send SBOM (default: false)"`
+}
+
+// attachmentState defines the state stored after attachment
+type attachmentState struct {
+	SendAttestation bool `json:"send_attestation"`
+	SendSBOM        bool `json:"send_sbom"`
+}
+
+// registrationState defines the state stored after registration
+type registrationState struct {
+	// No additional state needed for webhook besides the URL stored in credentials
+}
+
+// webhookPayload defines the JSON schema for the webhook payload
+type webhookPayload struct {
+	Metadata *sdk.ChainloopMetadata `json:"Metadata"`
+	Data     json.RawMessage        `json:"Data"` // e.g., SBOM or attestation JSON
+	Kind     string                 `json:"Kind"` // e.g., "SBOM_CYCLONEDX", "ATTESTATION"
+}
+
+// New initializes the webhook integration
+func New(l log.Logger) (sdk.FanOut, error) {
+	base, err := sdk.NewFanOut(
+		&sdk.NewParams{
+			ID:          "webhook",
+			Version:     "1.0",
+			Description: "Send Attestation and SBOMs to a generic webhook URL as JSON payloads",
+			Logger:      l,
+			InputSchema: &sdk.InputSchema{
+				Registration: registrationRequest{},
+				Attachment:   attachmentRequest{},
+			},
+		},
+		// Subscribe to SBOMs only, attestations are handled separately
+		sdk.WithInputMaterial(schemaapi.CraftingSchema_Material_SBOM_CYCLONEDX_JSON),
+		sdk.WithInputMaterial(schemaapi.CraftingSchema_Material_SBOM_SPDX_JSON),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create FanOut integration: %w", err)
+	}
+
+	return &Integration{
+		FanOutIntegration: base,
+		client:            &http.Client{},
+	}, nil
+}
+
+// Register is executed when registering the webhook integration
+func (i *Integration) Register(_ context.Context, req *sdk.RegistrationRequest) (*sdk.RegistrationResponse, error) {
+	i.Logger.Info("registration requested")
+
+	// Parse the registration payload
+	var regReq registrationRequest
+	if err := sdk.FromConfig(req.Payload, &regReq); err != nil {
+		i.Logger.Errorw("failed to parse registration payload", "error", err)
+		return nil, fmt.Errorf("invalid registration request: %w", err)
+	}
+
+	// Validate the URL
+	if err := validateURL(regReq.URL); err != nil {
+		i.Logger.Errorw("invalid webhook URL", "error", err, "url", regReq.URL)
+		return nil, fmt.Errorf("invalid webhook URL: %w", err)
+	}
+
+	// Optionally, perform a test request to ensure the webhook URL is reachable
+	if err := i.testWebhookURL(context.Background(), regReq.URL); err != nil {
```

Comment:
instead of `context.Background()` just start using the parent context in the function that's currently disabled `func (i *Integration) Register(_ context.Context, ...`